Fixes

inferno-chromium opened the pull request:
* Add reproduce command.
* Remove unneeded run and just_run command with duplicate code as run_fuzzer.
* Fix docs for reproduction.

mikea commented:
I suggest you put flag values into environment variable next to _OPTIONS and prepend those in run scripts we use. This way will keep it consistent through build system as well. WDYT?

inferno-chromium commented:
@mikea - the problem is reproduce script needs to be in base-builder for the compile command to work (if custom source_path is specified). I put in DockerFile for base-runner before but then we would need to duplicate it in both base-runner and base-builder.

inferno-chromium commented:
Last fix addresses the comment. I could split reproduce so that compile was part of build_fuzzers step. this fixed another bug which is when running reproduce, we don't trigger compile everytime. see reproducing.md change.

mikea commented on infra/helper.py:
```diff
@@ -264,6 +273,38 @@ def coverage(run_args):
+  parser = argparse.ArgumentParser('helper.py reproduce')
+  parser.add_argument('project_name', help='name of the project')
+  parser.add_argument('fuzzer_name', help='name of the fuzzer')
+  parser.add_argument('testcase_path', help='absolute path of local testcase')
```
I suggest you translate path to absolute instead. You use `~/Downloads` in examples yourself :)

inferno-chromium replied:
Looks like docker does the path translation automatically.

docker run --rm -i --cap-add SYS_PTRACE -v /usr/local/google/home/aarya/build/oss-fuzz/build/out/libarchive:/out -v /usr/local/google/home/aarya/Downloads/fuzz-3-libarchive_fuzzer:/testcase -t ossfuzz/base-runner reproduce /out/libarchive_fuzzer

inferno-chromium commented:
Removed the 'absolute' keyword since docker does the path translation automatically.

inferno-chromium commented:
My bad, i take it back, it only does expanduser, but not absolute path.
